### ISSUE TITLE
Hunter Pet Stable/pet Lost (During Switching talent)

### ISSUE THREAD

**Issue by mahbub2r** [SunwellTracker/issues#1362]
Description: 
Hunter Pet Stable is bug when you switch talent.  I lost like 4 pets including a rare pet. once I opened ticket and GM said you need to put the pet in the stable before switching talent. but its also not a good idea because if this is the case then pet stable have 4 slot and active 1 slot . and for switching pet u always need to keep 1 slot free ( because you cant keep the pet in an active slot, it will be lost) so hunter getting 1 less pet. another

How it works: 
Hunter Losing pet when they switch talent {Please check description)

How it should work: 
no matter what pet you have active BM pet or MM/SV pet once you switch talent pet will be in stable auto like dismissed and you will be needing call pet to get it. if a BM pet is active but we switched talent to MM pet call will not work but still pet will be in an active slot like dismissed, so hunter will open stable and switch pet by dragging.

Also if someone switch MM/SV to BM if they call pet while MM/SV pet was on active slot pet will come (As BM can have all MM/SV pet too) but with 2 additional talent points like you have 2 talent points left to spend as BM have 2 additional talent point than others spec for pet.

Source (you should point out proofs of your report, please give us some source):

**Comment by Zarant** [comment]
That's just a duplicate from #1236 the root cause of the issue is when you try to stable a pet when your pet isn't out. Whenever you switch talents, your pet is automatically dismissed, and if you swap stable slots, poof, your pet is gone.

**Comment by mahbub2r** [comment]
Yes, according to you this server system you cant have 5x pet that's the biggest proof. Because this not how its work. Hunter always can have 5x pet

**Comment by BestDKafrica** [comment]
Can confirm that it isnt working as intended.
Everytime I swap to Dual Specc, Sunwell will delete one of my pets from the stable randomly.

NEEDS A FIX
